src/components/Goals/BikeGoal/BikeGoalAlertManager.jsx

```diff
@@ -14,7 +17,9 @@ const BikeGoalAlertManager = () => {
     setShowGoals(false)
   }
   const onParticipate = () => {
-    // TODO
+    navigate('/bikegoal/onboarding', {
+      state: { background: location }
+    })
```

Comment:
Does it mean that you set the `current location` object (and so it's value) inside a state called `background`. 

And after that, when you call `navigate(location.state.background)` it goes to the value of this state? Its pretty nice! 

The end of the `returnUrl` inside the URI!

Reply:
c'est exactement ça ;) un expl ici https://github.com/cozy/coachCO2/blob/master/src/components/Views/BikeGoalAbout.jsx#L50